setup.py

```diff
@@ -152,7 +152,8 @@ def default_f2py():
     """
     interpdir, interp = os.path.split(sys.executable)
     if interp[0:6] == 'python':
-        suffixes = [interp[6:], '-' + interp[6:]]
+        vers = "{0.major:01d}.{0.minor:01d}".format(sys.version_info)
+        suffixes = [interp[6:], '-' + interp[6:], vers, '-' + vers]
         if '.' in interp[6:]: #try slicing off suffix-of-suffix (e.g., exe)
             suffix = interp[6:-(interp[::-1].index('.') + 1)]
             suffixes.extend([suffix, '-' + suffix])
```

Comment:
The new version-based stuff should come after the eliminate-suffix-of-suffix stuff. This is both to keep things together (there's the "our interpreter has a weird suffix and we should check that" in two flavors, and then the "try the version) and also because the interpreter suffix should be checked before the version one.

So if you have a Python interpreter version 3.7 in python-weirdbuild.exe and the associated f2py is f2py-weirdbuild, you want to get f2py-weirdbuild whereas as written you'd get f2py-3.7 if it happened to exist. You probably wouldn't want that.

i.e. move your two lines after the "if" block and make it a `suffixes.extend`.

Reply:
Makes sense; change made.